examples/response-metrics/apicast_response_metrics.lua

```diff
@@ -0,0 +1,66 @@
+--- Custom APICAST_MODULE which overrides post_action from original
+--- https://github.com/3scale/apicast/blob/3.0-stable/apicast/src/apicast.lua
+
+-- load and initialize the parent module
+local apicast = require('apicast').new()
+
+local _M = { _VERSION = '3.0.0', _NAME = 'APIcast with response metrics' }
+local mt = { __index = setmetatable(_M, { __index = apicast }) }
+
+function _M.new()
+  return setmetatable({}, mt)
+end
+
+local function send_count_metrics()
+  -- get the document count from the custom response header
+  local document_count = ngx.resp.get_headers()['x-document-count'];
+  local user_key = ngx.req.get_headers()['user_key'];
+
+  -- only report metrics if the response was a success and the custom header exists
+  if ngx.status == ngx.HTTP_OK and document_count then
+    ngx.log(ngx.INFO, '[3scale-metrics] sending metric to 3scale document-count: ', document_count)
+    local report = ngx.location.capture("/report_metric",
```

Comment:
@nburkley as @mikz pointed out it would be better to do this in lua. You could make 2 requests here to 3scale, first an authorise and then a report. This is done in the post_action phase and doesn't impact the latency on the client request. If the authorize returns a 409 you can allow the call through but if it returns a 403 then the call should be rejected, then you can report based on the authorize response. The authorize already exists in the backend_client so you would just need to add the report call there.

Reply:
Sure, I can do that - I was trying to avoid modifying the [backend_clint](https://github.com/3scale/apicast/blob/master/apicast/src/backend_client.lua) code - hence the call to an nginx conf.

I didn't have any auth-checks as I as going with the assumption that if the request made it thorugh our stack successfully it was already authorized - I can add this though.